File: wordNet/src/main/java/org/sqlunet/wordnet/style/Colors.java
/*
 * Copyright (c) 2019. Bernard Bou <1313ou@gmail.com>.
 */

package org.sqlunet.wordnet.style;

import android.content.Context;
import android.graphics.Color;

import org.sqlunet.wordnet.R;

import androidx.annotation.NonNull;

/**
 * Color values
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public class Colors
{
	static public int membersBackColor = Color.TRANSPARENT;
	static public int membersForeColor = Color.TRANSPARENT;

	static public int wordBackColor = Color.TRANSPARENT;
	static public int wordForeColor = Color.TRANSPARENT;

	static public void setColorsFromResources(@NonNull final Context context)
	{
		// do not reorder : dependent on resource array order

		int[] palette = context.getResources().getIntArray(R.array.palette_wn);
		int i = 0;
		membersBackColor = palette[i++];
		membersForeColor = palette[i++];

		wordBackColor = palette[i++];
		wordForeColor = palette[i++];
	}
}

File: wordNet/src/main/java/org/sqlunet/wordnet/style/Colors.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.wordnet.style

import android.content.Context
import android.graphics.Color
import org.sqlunet.wordnet.R

/**
 * Color values
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
object Colors {

    var membersBackColor = Color.TRANSPARENT

    var membersForeColor = Color.TRANSPARENT

    var wordBackColor = Color.TRANSPARENT

    var wordForeColor = Color.TRANSPARENT

    fun setColorsFromResources(context: Context) {
        // do not reorder : dependent on resource array order
        val palette = context.resources.getIntArray(R.array.palette_wn)
        var i = 0
        membersBackColor = palette[i++]
        membersForeColor = palette[i++]
        wordBackColor = palette[i++]
        wordForeColor = palette[i]
        // increment i if colors added
    }
}
